go/pkg/client/client.go

```diff
@@ -43,6 +43,60 @@ const (
 	HomeDirMacro = "${HOME}"
 )
 
+// AuthType indicates the type of authentication being used.
+type AuthType int
+
+const (
+	// Unknown auth type.
+	UnknownAuth AuthType = iota
+
+	// No authentication used to connect to the RBE service.
+	NoAuth
+
+	// External auth token used to connect to the RBE service.
+	ExternalTokenAuth
+
+	// A JSON credentials file used to connect to the RBE service.
+	CredsFileAuth
+
+	// Application default credentials used to connect to the RBE service.
+	ApplicationDefaultCredsAuth
+
+	// GCE machine credentials used to connect to the RBE service.
+	GCECredsAuth
+)
+
+// ClientInitError is used to wrap the error returned when initializing a new
+// client to also indicate the type of authentication used.
+type ClientInitError struct {
+	// Err refers to the underlying client initialization error.
+	Err error
+
+	// AuthUsed stores the type of authentication used to connect to RBE.
+	AuthUsed AuthType
+}
+
+// Error returns a string error that includes information about the
+// type of auth used to connect to RBE.
+func (ce *ClientInitError) Error() string {
+	var at string
+	switch ce.AuthUsed {
+	case UnknownAuth:
+		at = "unknown authentication type"
+	case NoAuth:
+		at = "no authentication"
+	case ExternalTokenAuth:
+		at = "external authentication token (gcert?)"
+	case CredsFileAuth:
+		at = "credentials file"
+	case ApplicationDefaultCredsAuth:
+		at = "application default credentials"
+	case GCECredsAuth:
+		at = "gce credentials"
+	}
```

Comment:
nit: this could be actually a part of `AuthType` enum definition (`func (s AuthType) String() string`)

Reply:
Done!